spec/index.md

```diff
@@ -1550,6 +1557,16 @@ If this traps (e.g. runs out of cycles), the state changes from the `cleanup` fu
 
 There must be at most one call to `ic0.call_on_cleanup` between `ic0.call_new` and `ic0.call_perform`.
 
+-   `ic0.call_set_timeout : (seconds : i32) -> ()`
+
+    Requests the system to generate a synthetic timeout reject if no reply is received within the specified amount of seconds. The returned reject code will be `SYS_TIMEOUT`.
+
+    The `seconds` parameter is only advisory. First, it is silently bounded by the `MAX_CALL_TIMEOUT` system constant; i.e., larger timeouts are treated as equivalent to `MAX_CALL_TIMEOUT` and do not cause an error. Second, while the system will attempt not to time out the call too early, it may in exceptional circumstances trigger the timeout earlier than requested; the calling canister should not rely on the timeout not occurring early. Finally, even under normal circumstances, the timeout may trigger significantly later than the specified time.
```

Comment:
Could you maybe comment on why we need the
```
Finally, even under normal circumstances, the timeout may trigger significantly later than the specified time.
```
Is it just to give the implementation more flexibility, or is there a concrete concern that requires this formulation?

Reply:
I was thinking of something like another DTS message handler already being scheduled; I don't know how we'd prioritize that, especially if we have several of those. In the extreme case, if you simultaneously make 10 calls that time out, and each timeout message handler does several DTS rounds, you could end up triggering the timeout significantly later (even though the system is running normally).

Reply:
Good point.

Another reason is probably also the subnet time jumping in bigger increments than expected because of some issues leading to lower FR or similar. More generally, one could probably say that timeouts will be at most as fine grained as the increments in terms of subnet time.

Reply:
Technically, if we're simply considering the effect of the subnet/canister being backlogged (as opposed e.g. optimizations in how we trigger timeouts that might result in timeouts only being triggered once ever N seconds), the timeout will still trigger on time. It's just that we cannot guarantee its immediate execution. Which is true for every kind of execution (including timers), so I would rather leave it out.